website/server/libs/user/index.js

```diff
@@ -112,7 +112,22 @@ export async function update (req, res, { isV3 = false }) {
     const newName = req.body['profile.name'];
     if (newName === null) throw new BadRequest(res.t('invalidReqParams'));
     if (newName.length > 30) throw new BadRequest(res.t('displaynameIssueLength'));
-    if (nameContainsSlur(newName)) throw new BadRequest(res.t('displaynameIssueSlur'));
+    if (stringContainsSlur(newName)) {
+      user.flags.chatRevoked = true;
+      await user.save();
```

Comment:
@Alys I read the original issue but I'm a bit confused about this part:

> If they do, the change should not be saved. If a slur is used, the user's public chat permissions should be revoked and moderators should be notified in the same way that they are for chat messages that contain slurs.

If the change is not saved the user won't have a slur in their username, so is it still necessary to revoke their chat permissions?

Reply:
Also I'm slightly concerned about the fact that we change an user property but return an error, so the user  won't see the fact that they're chat privileges have been revoked until they refresh the website.

Reply:
> > If a slur is used, the user's public chat permissions should be revoked

That's necessary because the user might go on to use a version of the slur that has one letter replaced to avoid the blocker and could then post to any public chat. Basically, if a player tries to use a slur in their name or profile, it's a danger sign for us and we wouldn't want them to have chat permissions until they'd emailed us and we could get them to agree in an email to follow the Community Guidelines.

> Also I'm slightly concerned about the fact that we change an user property but return an error, so the user won't see the fact that they're chat privileges have been revoked until they refresh the website.

That's a great point! We could change the error messages so that they contain more detail, for example using words from the error message they see if they try to post a slur to chat. If that's okay with you, I'll put some words together and run them past beffymaroo and the other mods.

Reply:
Sounds good to me